src/main/java/com/forgeessentials/api/UserIdent.java

```diff
@@ -214,39 +271,111 @@ public static synchronized UserIdent get(EntityPlayerMP player)
         else
         {
             String name = player.getCommandSenderName();
-            if (name != null && !name.equals(ident.username))
+
+            String newname = name + "_" + ident.uuid.hashCode();
+            if (name != null && !name.equals(ident.username) && !newname.equals(ident.username))
             {
-                byUsername.remove(ident.username);
+                if (byUsername.containsKey(name.toLowerCase()))
+                {
+                    try
+                    {
+                        throw new Throwable();
```

Comment:
That's not how you generate a stack trace!
Use a simple `new Exception().printStackTrace()` instead.